### ISSUE TITLE
Multilingual Bert

### ISSUE THREAD

**Issue by emanueledirosa** [amaiya/ktrain#18]
That's a great project!
However, It is not clear which BERT pre-trained model version is loaded.
Is there a way to specify the multilingual version? or even between Base/Large and Cased/Uncased?

For chinese, here just "bert" is specified in both preprocessor and in the text_classifier:
https://github.com/amaiya/ktrain/blob/master/examples/text/ChineseHotelReviews-BERT.ipynb

I'm interested to use the "BERT-Base Multilingual Cased" from  here: https://github.com/google-research/bert
Thank you.
Regards,
Emanuele

**Comment by amaiya** [comment]
Thanks, Emanuele.   

I decided to make the selection of the BERT model automatic based on the language detected. The language is detected based on a small random sample of your training set.  If English is detected, then `uncased_L-12_H-768_A-12` is downloaded.  If Chinese is detected, then `chinese_L-12_H-768_A-12` is downloaded.  If any other language is detected, then `multi_cased_L-12_H-768_A-12` (i.e., BERT-Base, Multilingual Cased) is downloaded.

If you're using a language that's not English or Chinese, then BERT-Base, Multilingual Cased is used.  So, you should be fine.

If not, then there are two "hacks" to use the multilingual model.


1. You can set the language (i.e., lang) explicitly in any of the `texts_from*` functions. If anything other than `en` or `zh-*`  is specified, then the multilingual cased model is automatically used for BERT. So, if you wanted to use the multilingual model for English, maybe just try entering something other than `en` (and `zh-*`) for `lang` parameter. I  haven't actually tried this, but it should work.
2. The models are downloaded to `~/ktrain_data`.  As a hack, if you really need to use a model other than the one given by the above scheme, you can always just symlink to it.  For instance, if you wanted to use the multilingual model for Chinese (instead of `chinese_L-12_H-768_A-12`, which is the recommended model), you can just create a symlink called `chinese_L-12_H-768_A-12`  that points to the multilingual model folder.  I actually did this initially for testing purposes and it is confirmed to work.   I suppose the same is true for Base vs. Large models, but this hasn't been tested.

Thanks.

**Comment by sophiegroenwold** [comment]
Hello Arun, 

Thanks for elaborating on this. I've been using your symlink "hack" to specify models when I want to go against the default scheme you specified. You've noted here that the symlink method is confirmed to work; however, is there any way to verify what model is downloaded? For instance, if I wanted to use the multilingual model for Chinese and I had a symlink from `chinese_L-12_H-768_A-12` to the multilingual folder, is there some way to print the downloaded model to confirm that it is actually using the multilingual model instead of the Chinese model?

Thanks again.

**Comment by amaiya** [comment]
Hi @sophiegroenwold ,

The symlink hack was only needed for older versions of *ktrain*.

There are actually two ways to access BERT models in *ktrain* now.  The original API (described in [BERT Text Classification in 3 Lines of code](https://towardsdatascience.com/bert-text-classification-in-3-lines-of-code-using-keras-264db7e7a358)) and the newer API to Hugging Face transformers (described in [Text Classification with Hugging Face Transformers in TensorFlow 2](https://towardsdatascience.com/text-classification-with-hugging-face-transformers-in-tensorflow-2-without-tears-ee50e4f3e7ed)).  

For the newer Hugging Face transformer API ( recommended if you need more control over the model being used), you can explicitly specify any of the [pretrained models listed here](https://huggingface.co/transformers/pretrained_models.html) in addition to [community-uploaded models](https://huggingface.co/models).

So, for instance, if you want to use the multilingual model for Chinese training set, you can specify the model explicitly like this:
```python
import ktrain
from ktrain import text
MODEL_NAME = 'multi_cased_L-12_H-768_A-12'
t = text.Transformer(MODEL_NAME, maxlen=500, class_names=label_list)
trn = t.preprocess_train(x_train, y_train)
val = t.preprocess_test(x_test, y_test)
model = t.get_classifier()
learner = ktrain.get_learner(model, train_data=trn, val_data=val, batch_size=6)
learner.fit_onecycle(3e-5, 1)
```
The `x_train` and `y_train` arrays might look something like this (except the text will be Chinese in your case):
```python
x_train = ['I hate this movie.', 'I like this movie.']
y_train = ['negative', 'positive']
x_test = ['I despise this movie.', 'I love this movie.']
y_test = ['negative', 'positive']
```
The `class_names` argument is not needed if the `y_train` and `y_test` are string labels like in this example.

You take a look at the [article describing this newer API](https://towardsdatascience.com/text-classification-with-hugging-face-transformers-in-tensorflow-2-without-tears-ee50e4f3e7ed) I mentioned above for more details.   Hope this helps.

**Comment by justinchuntingho** [comment]
First of all, thanks for the amazing package! I have a question while following the instruction on the newer transformers API. In the older `texts_from*` API we could specify ngram_range, I am wondering if the transformers API can do the same?
